zerver/migrations/0122_disallow_markdown_chars_in_stream_and_user_name.py

```diff
@@ -0,0 +1,49 @@
+# -*- coding: utf-8 -*-
+from django.db import models, migrations
+from django.db.backends.postgresql_psycopg2.schema import DatabaseSchemaEditor
+from django.db.migrations.state import StateApps
+
+def remove_special_chars_from_streamname(apps, schema_editor):
+    # type: (StateApps, DatabaseSchemaEditor) -> None
+    Stream = apps.get_model('zerver', 'Stream')
+    NAME_INVALID_CHARS = ['*', '@', '`', '#']
+    for entry in Stream.objects.all():
+        stream_name = entry.name
+
+        if (set(stream_name).intersection(NAME_INVALID_CHARS)):
+            for char in NAME_INVALID_CHARS:
+                stream_name = stream_name.replace(char, ' ')
+
+        entry.name = entry.name.replace(entry.name, stream_name, 1)
+        entry.save()
```

Comment:
A bunch of small comments. I'd write this as (untested):
```
Stream = apps.get_model('zerver', 'Stream')
for stream in Stream.objects.all():
    if (set(stream.name).intersection(Stream.NAME_INVALID_CHARS)):
        for char in Stream.NAME_INVALID_CHARS:
            stream.name = stream.name.replace(char, ' ').strip()
        stream.save(update_fields='name')
```

A few things to note:
* entry -> stream
* don't need to introduce a new variable stream_name; using stream.name is better
* I believe we have access to Stream.NAME_INVALID_CHARS, though might want to double check, now that I think about it :).
* note the .strip() so we don't turn `#streamname` into ` streamname`
* `save()` takes an update_fields argument, which is a performance benefit (it only saves those fields)
* most importantly, moving the save() to inside the if statement means we only need to run the save if something has changed.

Reply:
hm, another remaining thing to do here is check for duplicates. I.e. before the stream.save line, I would do something like
```
while Stream.objects.filter(name__iexact=stream.name, realm=stream.realm).exists():
   stream.name = stream.name + '^'   
   if len(stream.name) > 60: 
       # extremely unlikely, so just do something valid
        stream.name = stream.name[-60:]
```

(Can replace `^` with a different character if you have a better idea)

Reply:
@rishig Thankyou for your review! I have made the changes.
(Also I checked that we don't have access to Stream.NAME_INVALID_CHARS, we get an AttributeError.)